libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -103,7 +103,10 @@ internal class MapboxTripSession(
             RoutesExtra.ROUTES_UPDATE_REASON_REFRESH -> {
                 if (routes.isNotEmpty()) {
                     val primaryRoute = routes.first()
-                    val alternatives = navigator.refreshRoute(primaryRoute)
+                    val alternatives = navigator.refreshRoute(
+                        primaryRoute,
+                        routeProgress?.currentRouteGeometryIndex
```

Comment:
when `routeProgress` can be null there?

Reply:
`routeProgress` is nullable https://github.com/mapbox/mapbox-navigation-android/blob/c68ad1e8d03ec42355277d4f75521c732903a947/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L181 Are you suggesting using `!!` instead?

Reply:
nope. I'm trying to understand if the case when the SDK refreshes a route without running a trip session is possible and if should we support this scenario. WDYT?

Reply:
`updateRouteProgress` receives `RouteProgress?` so technically it can be null even if the session is running. But it does seem like a weird case. I could add the check and ignore null route progress. But since in the end there are two methods: with index and without it, maybe we should leave it as-is? Just in case there can be (now or in the future) some not-so-obvious situations when `routeProgress` is null.

Reply:
What's the point of passing current route geometry index? NN can just update all annotations as the java code has already merged refreshed annotations to current, taking current route geometry index into account. What's more, we can pass a wrong geometry index to the navigator as we use current one in router wrapper and trip session while user is moving.